crates/revm/src/evm_impl.rs

```diff
@@ -91,6 +91,52 @@ impl<'a, GSPEC: Spec, DB: Database, const INSPECT: bool> EVMImpl<'a, GSPEC, DB,
         }
         Ok(())
     }
+
+    /// If the transaction is not a deposit transaction, subtract the L1 data fee from the
+    /// caller's balance directly after minting the requested amount of ETH.
+    #[cfg(feature = "optimism")]
+    fn remove_l1_cost(
+        is_deposit: bool,
+        tx_caller: B160,
+        tx_l1_cost: Option<U256>,
+        db: &mut DB,
+        journal: &mut JournaledState,
+    ) -> Result<(), EVMError<DB::Error>> {
+        if is_deposit {
+            return Ok(());
+        }
+        if let Some(l1_cost) = tx_l1_cost {
+            let acc = journal
+                .load_account(tx_caller, db)
+                .map_err(EVMError::Database)?
+                .0;
+            let res = acc.info.balance.saturating_sub(l1_cost);
```

Comment:
Ideally this should throw an underflow error, maybe `InvalidTransaction::LackOfFundForMaxFee`?

Reply:
Agreed, this doesn't make sense - my oversight.